src/app/core/form-dialog/form-dialog.service.ts

```diff
@@ -24,14 +26,27 @@ export class FormDialogService {
     private confirmationDialog: ConfirmationDialogService
   ) {}
 
-  openDialog<T extends ShowsEntity>(
+  openDialog<
+    E extends Entity,
+    T extends ShowsEntity<E> | (ShowsEntity<E> & OnInitDynamicComponent)
+  >(
     entityDetailsComponent: ComponentType<T>,
-    entity: any
+    entity: E,
+    componentConfig?: any
   ): MatDialogRef<T> {
     const dialogRef = this.dialog.open(entityDetailsComponent, {
       width: "80%",
     });
+
     dialogRef.componentInstance.entity = entity;
+    if (
+      typeof (dialogRef.componentInstance as OnInitDynamicComponent)
+        .onInitFromDynamicConfig === "function"
+    ) {
+      (dialogRef.componentInstance as OnInitDynamicComponent).onInitFromDynamicConfig(
```

Comment:
Maybe a test that ensures that the function is actually called when an appropriate component is passed

Reply:
added in the core PR https://github.com/Aam-Digital/ndb-core/pull/604